src/phone-number-field.tsx

```diff
@@ -21,14 +26,39 @@ type InputProps = Omit<React.InputHTMLAttributes<HTMLInputElement>, 'value' | 'o
     onInput?: (event: React.FormEvent<HTMLInputElement>) => void;
     prefix?: string;
     e164?: boolean;
+    format?: NumberFormatter;
 };
 
 const isValidPrefix = (prefix: string): boolean => !!prefix.match(/^\+\d+$/);
 
-const PhoneInput = ({inputRef, value, defaultValue, onChange, prefix, e164, ...other}: InputProps) => {
+const PhoneInput = ({
+    inputRef,
+    value,
+    defaultValue,
+    onChange,
+    prefix,
+    e164,
+    format: formatFromProps,
+    ...other
+}: InputProps) => {
     const [selfValue, setSelfValue] = React.useState(defaultValue ?? '');
     const ref = React.useRef<HTMLInputElement | null>(null);
     const {i18n} = useTheme();
+    const formatRef = React.useRef<NumberFormatter>(formatFromProps || formatPhoneDummy);
+    /**  this state is used to force a re-render when libphonenumber is loaded */
+    const [isLibphonenumberLoaded, setIsLibphonenumberloaded] = React.useState(false);
+
+    React.useEffect(() => {
+        if (formatFromProps) {
+            formatRef.current = formatFromProps;
+        } else {
+            import('@telefonica/libphonenumber' /* webpackChunkName: "libphonenumber" */).then((lib) => {
+                libphonenumber = lib;
+                formatRef.current = formatPhoneUsingLibphonenumber;
+                setIsLibphonenumberloaded(true);
```

Comment:
what happens if the user has introduced some text before libphonenumber has loaded? is the text formatted once the lib has loaded? Can we have a test of this case?

Reply:
added a unit test to verify this case